RELEASE_CHECKLIST.md

```diff
@@ -8,30 +8,27 @@ More info about our release process can be found in the [`RELEASE_PROCESS.md`](.
 
 - [ ] Look through all open issues and PRs (if any) of that milestone and close them / move them to another
 milestone if still open
-- [ ] Look through all closed issues and PRs of that milestone to see what has changed. Run `./scripts/prs-since-last-tag` or if you want to run against a specific tag `./scripts/prs-since-last-tag v1.20.0` to get a list of all merged PR's since a specific tag
-- [ ] Close milestone on GitHub
-- [ ] Create a new draft release in GitHub
-
-# Testing
-
-- [ ] Create a Serverless service (with some events), deploy and test it intensively
-- [ ] Look through the milestone and test all of the new major changes
-- [ ] Run `npm test`
-- [ ] Run `npm run simple-integration-test`
-- [ ] Run `npm run complex-integration-test`
+- [ ] Create a new branch for the release
+- [ ] Bump the version number in `package.json`
+- [ ] Run `./scripts/prs-since-last-tag <OLD-TAG>`
+- [ ] Save the terminal output to your clipboard
+- [ ] Close the milestone on GitHub
+- [ ] Create a new [**draft** release](https://github.com/serverless/serverless/releases/new) in GitHub
+  - [ ] Use the content in your clipboard as a description (without the heading)
+  - [ ] Ensure that the "Tag version" follows our naming convention
 
 ## Prepare Package
 
-- [ ] Create a new branch to bump version in `package.json`
 - [ ] Install the latest `npm` version or Docker container with latest `node` and `npm`
-- [ ] Bump version in `package.json`, remove `node_modules` folder and run `npm install` and `npm prune --production && npm shrinkwrap`
-- [ ] Look through closed PRs and update `CHANGELOG.md`
+- [ ] Remove the `node_modules` folder and the `package-lock.json` file and run `npm install`
```

Comment:
We may add clarification comment:

_(Removing both ensures that `package-lock.json` is updated with latest versions of dependencies)_